controller/tracker.go

```diff
@@ -63,96 +76,150 @@ func (c *TrackerController) Create(ctx *app.CreateTrackerContext) error {
-	return result
+	return ctx.OK([]byte{})
 }
 
 // Show runs the show action.
 func (c *TrackerController) Show(ctx *app.ShowTrackerContext) error {
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
 	return application.Transactional(c.db, func(appl application.Application) error {
-		t, err := appl.Trackers().Load(ctx.Context, ctx.ID)
+		tracker, err := appl.Trackers().Load(ctx.Context, id)
 		if err != nil {
-			cause := errs.Cause(err)
-			switch cause.(type) {
-			case remoteworkitem.NotFoundError:
-				log.Error(ctx, map[string]interface{}{
-					"tracker_id": ctx.ID,
-				}, "tracker controller not found")
-				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrNotFound(err.Error()))
-				return ctx.NotFound(jerrors)
-			default:
-				jerrors, httpStatusCode := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInternal(err.Error()))
-				return ctx.ResponseData.Service.Send(ctx.Context, httpStatusCode, jerrors)
-			}
+			log.Error(ctx, map[string]interface{}{
+				"err":        err,
+				"tracker_id": ctx.ID,
+			}, "unable to load the tracker by ID")
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		return ctx.OK(t)
+		result := &app.TrackerSingle{
+			Data: convertTracker(appl, ctx.Request, *tracker),
+		}
+		return ctx.OK(result)
 	})
 }
 
 // List runs the list action.
 func (c *TrackerController) List(ctx *app.ListTrackerContext) error {
-	exp, err := query.Parse(ctx.Filter)
-	if err != nil {
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrBadRequest(fmt.Sprintf("could not parse filter: %s", err.Error())))
-		return ctx.BadRequest(jerrors)
-	}
-	start, limit, err := parseLimit(ctx.Page)
-	if err != nil {
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrBadRequest(fmt.Sprintf("could not parse paging: %s", err.Error())))
-		return ctx.BadRequest(jerrors)
-	}
 	return application.Transactional(c.db, func(appl application.Application) error {
-		result, err := appl.Trackers().List(ctx.Context, exp, start, &limit)
+		trackers, err := appl.Trackers().List(ctx)
 		if err != nil {
-			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInternal(fmt.Sprintf("Error listing trackers: %s", err.Error())))
-			return ctx.InternalServerError(jerrors)
+			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		return ctx.OK(result)
+		res := &app.TrackerList{}
+		res.Data = ConvertTrackers(appl, ctx.Request, trackers)
+		return ctx.OK(res)
 	})
 
 }
 
+// ConvertTrackers from internal to external REST representation
+func ConvertTrackers(appl application.Application, request *http.Request, trackers []remoteworkitem.Tracker) []*app.Tracker {
+	var ls = []*app.Tracker{}
+	for _, i := range trackers {
+		ls = append(ls, convertTracker(appl, request, i))
+	}
+	return ls
+}
+
 // Update runs the update action.
 func (c *TrackerController) Update(ctx *app.UpdateTrackerContext) error {
+	_, err := login.ContextIdentity(ctx)
```

Comment:
May be yes. This needs some more discussion. Since this is pr is about restructuring existing code, I wouldnot handle that in this PR. We can talk more about this and I can raise another PR for adding authentication. Does this sound good @pranavgore09 ?

Reply:
Currently the remote tracker controller is not mounted (see main.go). We disabled it until we add some authorization to this REST API.
So, it's OK to refactor this code without adding authorization if the controller remains unmounted.